### DISCUSSION TITLE
Business rules Execution

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3408] [Questions]
I think I have asked a similar question before : )

Today I am working on a simple object with seven properties. All of them are required.
When all the properties are present, a final business rule needs to run that does calculations based on all the properties.
However, when I test the business object with, say,  three properties missing, it still tries to run the final 'calculation' business rule.

Kind Regards

**Comment by danielmartind** [comment]
You can create a rule with a lower priority than the calculation rule that checks if the object is valid and stops processing if not.

```
  public class StopIfNotValid : PropertyRule
{
    public StopIfNotValid(IPropertyInfo primaryProperty) : base(primaryProperty)
    {
    }

    protected override void Execute(IRuleContext context)
    {
        if (!((BusinessBase)context.Target).IsValid)
        {
            context.AddSuccessResult(true);
        }
    }
}
```

**Reply by Chicagoan2016** [reply]
Thank you @danielmartind .
Since my 'calculation' rule requires multiple properties. I slightly modified your rule so that no business rules would be run for the 'whole' object in cases where the object is invalid but no luck : (
Although the 'StopIfNotValid' is getting called but it's causing short circuiting for the 'whole' object.

Here is my version, I just removed the primary property
```cs
  public class StopIfNotValid : PropertyRule
{
    public StopIfNotValid() : base()
    {
    }

    protected override void Execute(IRuleContext context)
    {
        if (!((BusinessBase)context.Target).IsValid)
        {
            context.AddSuccessResult(true);
        }
    }
}
```

**Reply by danielmartind** [reply]
Not sure of the details of your scenario.
I have 3 numbers and a total property with a rule that calulates the property. 

```
        BusinessRules.AddRule(new MinValue<int>(Num1Property, 5));
        BusinessRules.AddRule(new MinValue<int>(Num2Property, 10));
        BusinessRules.AddRule(new MinValue<int>(Num3Property, 15));

        BusinessRules.AddRule(new StopIfNotValid(TotalProperty));
        BusinessRules.AddRule(new CalcSum(TotalProperty, Num1Property, Num2Property, Num3Property) { Priority = 10 });
```

CalcSum only works when the three number properties are valid

**Reply by Chicagoan2016** [reply]
@danielmartind  I wonder if you try to set Num1 or Num2 from 'UI', will it trigger CalcSum?
In my case invalid dependent properties like Num1, Num2 are triggering the CalcSum

**Reply by danielmartind** [reply]
I've just set this up in a console app. Total doesn't change until all number properties are valid. You can see [here](https://github.com/danielmartind/CslaRuleTest)
